docs/develop/typescript/cancellation.mdx

```diff
@@ -383,3 +383,20 @@ export async function shieldAwaitedInRootScope(): Promise<any> {
 ```
 
 <!--SNIPEND-->
+
+## Cancel an Activity from a Workflow {#cancel-an-activity}
+
+Canceling an Activity from within a Workflow requires that the Activity Execution sends Heartbeats and sets a Heartbeat Timeout.
+If the Heartbeat is not invoked, the Activity cannot receive a cancellation request.
+When any non-immediate Activity is executed, the Activity Execution should send Heartbeats and set a [Heartbeat Timeout](/encyclopedia/detecting-activity-failures#heartbeat-timeout) to ensure that the server knows it is still working.
+
+When an Activity is canceled, an error is raised in the Activity at the next available opportunity.
+If cleanup logic needs to be performed, it can be done in a `finally` clause or inside a caught cancel error.
```

Comment:
I'd specify the error if there's a parent cancelation error (not sure there is in typescript) and link to the API docs for it. If this doesn't apply, no worries.

Reply:
Also, who is catching the error? The Workflow or the Activity if it isn't in the finally clause?